### PR TITLE
Handle certificates that end in newline

### PR DESCRIPTION
**Pull request by haydentherapper**
This fixes a bug where certificates with trailing newlines would not be parseable.

Fixes https://github.com/sigstore/cosign/issues/2237

Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

#### Release Note

Fixed bug when unmarshalling certificates with trailing newline

### CODE REVIEW

**Review by cpanato — DISMISSED** [review]

File: pkg/cryptoutils/certificate_test.go [hunk 1]
```diff
@@ -126,6 +126,12 @@ func TestCertificatesFromPEM(t *testing.T) {
 
 			expected: []*x509.Certificate{cert1},
 		},
+		{
+			name:     "one cert with trailing newline",
```
Comment (znewman01) [inline pkg/cryptoutils/certificate_test.go:130]: Can you add a test for two certs in a row with a bonus newline in between?

**Review by znewman01 — CHANGES_REQUESTED** [review]
1 minor request for a test case. Otherwise, looks great 🙌

Also, can you make sure that the CPython signatures work when you compile this into Cosign? https://www.python.org/download/sigstore/

File: pkg/cryptoutils/certificate_test.go (hunk 1 above)
Comment (haydentherapper) [reply pkg/cryptoutils/certificate_test.go:130]: yep, adding test. This worked previously because pem.Decode handles a newline at the start of a PEM block

**Comment by haydentherapper** [comment]
```
$ COSIGN_EXPERIMENTAL=1 cosign verify-blob --signature /tmp/Python-3.10.7.tgz.sig --certificate /tmp/Python-3.10.7.tgz.crt /tmp/Python-3.10.7.tgz
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tuf: warning using deprecated ecdsa hex-encoded keys
tlog entry verified with uuid: eef13c27c14927ab6027ccc993cd2c688f611e7ac08649b5c197f49144d9b0e2 index: 3404126
Verified OK
```

**Review by znewman01 — APPROVED** [review]

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**